### PR TITLE
dead links flag

### PR DESCRIPTION
**Pull request by allandequeiroz**
Sending a big pull request because is just a flag after links. Should be ok to review.

### CODE REVIEW

File: en/how-to-navigate/contribute_to_this_book.md [hunk 1]
```diff
@@ -8,6 +8,6 @@ clone the repo **https://www.gitbook.com/book/adoptopenjdk/adoptopenjdk-getting-
 
 or 
 
-You can also [contact](https://www.gitbook.com/book/adoptopenjdk/adoptopenjdk-getting-started-kit/contact) the authors with your github username to be added to the repo.
+You can also [contact](https://www.gitbook.com/book/adoptopenjdk/adoptopenjdk-getting-started-kit/contact) [dead-link] the authors with your github username to be added to the repo.
```
Comment (neomatrix369) [inline en/how-to-navigate/contribute_to_this_book.md]: https://adoptopenjdk.gitbooks.io/adoptopenjdk-getting-started-kit/content/en/feedback.html new link

File: en/intermediate-steps/code_analysis_of_openjdk_projects.md [hunk 2]
```diff
@@ -12,7 +12,7 @@ https://java.net/projects/adoptopenjdk/pages/JavaSourcesWithIntellij
 
 After going through the steps below, we'll have this dashboard:
 
-![[[SonarQube Configured for OpenJDK8 and OpenJDK9]]](SonarQube-OpenJDK8-and-OpenJDK9.png)
+![[[SonarQube Configured for OpenJDK8 and OpenJDK9]]](SonarQube-OpenJDK8-and-OpenJDK9.png) [dead-link]
```
Comment (neomatrix369) [inline en/intermediate-steps/code_analysis_of_openjdk_projects.md]: Nudge me about this, we should have this somewhere

File: en/intermediate-steps/code_analysis_of_openjdk_projects.md [hunk 3]
```diff
@@ -32,9 +32,9 @@ Then we'll run SonarQube analysis across the codebase, and create custom filters
 Use this [blog](https://neomatrix369.wordpress.com/2013/09/16/installing-sonarqube-formely-sonar-on-mac-os-x-mountain-lion-10-8-4/) to install SonarQube or the instructions below (instructions for Mac OS, might need tweaks for other OSes)
 
 **Install MySQL and SonarQube on the Ubuntu VM**
-1. Download and install MySQL on Ubuntu. See section titled "Installation" on https://help.ubuntu.com/12.04/serverguide/mysql.html
+1. Download and install MySQL on Ubuntu. See section titled "Installation" on https://help.ubuntu.com/12.04/serverguide/mysql.html [dead-link]
```
Comment (neomatrix369) [inline en/intermediate-steps/code_analysis_of_openjdk_projects.md]: We should look for an alternative link for this

File: en/intermediate-steps/code_analysis_of_openjdk_projects.md [hunk 4]
```diff
@@ -32,9 +32,9 @@ Then we'll run SonarQube analysis across the codebase, and create custom filters
...
 
-2. In MySQL, create the sonar database and user - https://github.com/SonarSource/sonar-examples/blob/master/scripts/database/mysql/create_database.sql
+2. In MySQL, create the sonar database and user - https://github.com/SonarSource/sonar-examples/blob/master/scripts/database/mysql/create_database.sql [dead-link]
```
Comment (neomatrix369) [inline en/intermediate-steps/code_analysis_of_openjdk_projects.md]: We should look for an alternative link for this

File: en/intermediate-steps/testing_java_early_project.md [hunk 5]
```diff
@@ -1,14 +1,14 @@
...
 But this does not need to stop here, you can select your own java library or opensource project, build it using JDK9 and then run it via JDK9 or JRE9.
 
 * Ensure you set your PATH and/or JAVA_HOME to point to the JDK image you built using OpenJDK9 (see build folder ```build/linux-x86_64-normal-server-release/images/jdk``` for a Linux build).
-* JDK 9 Early Access binary - download the Early Access binaries of Java 9 from https://jdk9.java.net/download/ (ensure you set your PATH and/or JAVA_HOME to point to the location of the downloaded JDK).
-* JDK 9 Jigsaw Early Access binary - download the Early Access binaries of Java 9 from https://jdk9.java.net/jigsaw/ (ensure you set your PATH and/or JAVA_HOME to point to the location of the downloaded JDK).
+* JDK 9 Early Access binary - download the Early Access binaries of Java 9 from https://jdk9.java.net/download/ [dead-link] (ensure you set your PATH and/or JAVA_HOME to point to the location of the downloaded JDK).
+* JDK 9 Jigsaw Early Access binary - download the Early Access binaries of Java 9 from https://jdk9.java.net/jigsaw/ [dead-link] (ensure you set your PATH and/or JAVA_HOME to point to the location of the downloaded JDK).
```
Comment (neomatrix369) [inline en/intermediate-steps/testing_java_early_project.md]: jdk.java.net/jigsaw/

File: en/intermediate-steps/testing_java_early_project.md [hunk 6]
```diff
@@ -1,14 +1,14 @@
...
+**Adopt OpenJDK runs its own efforts to encourage and support the above efforts, see https://java.net/projects/adoptopenjdk/pages/TestingJava8.**  [dead-link]
 
 But this does not need to stop here, you can select your own java library or opensource project, build it using JDK9 and then run it via JDK9 or JRE9.
 
 * Ensure you set your PATH and/or JAVA_HOME to point to the JDK image you built using OpenJDK9 (see build folder ```build/linux-x86_64-normal-server-release/images/jdk``` for a Linux build).
-* JDK 9 Early Access binary - download the Early Access binaries of Java 9 from https://jdk9.java.net/download/ (ensure you set your PATH and/or JAVA_HOME to point to the location of the downloaded JDK).
-* JDK 9 Jigsaw Early Access binary - download the Early Access binaries of Java 9 from https://jdk9.java.net/jigsaw/ (ensure you set your PATH and/or JAVA_HOME to point to the location of the downloaded JDK).
+* JDK 9 Early Access binary - download the Early Access binaries of Java 9 from https://jdk9.java.net/download/ [dead-link] (ensure you set your PATH and/or JAVA_HOME to point to the location of the downloaded JDK).
```
Comment (neomatrix369) [inline en/intermediate-steps/testing_java_early_project.md]: http://jdk.java.net/9/

File: en/openjdk-projects/kulla/kulla-pom-xml.md [hunk 7]
```diff
@@ -151,7 +151,7 @@ And then run the following maven command:
 $ mvn clean package -Dbuild.image.folder=$JAVA9_HOME
 ```
 
-where ```$JAVA9_HOME``` points to the JDK9 binary downloaded from the [JDK9 Early Access site](https://jdk9.java.net/download/).
+where ```$JAVA9_HOME``` points to the JDK9 binary downloaded from the [JDK9 Early Access site](https://jdk9.java.net/download/) [dead-link].
```
Comment (neomatrix369) [inline en/openjdk-projects/kulla/kulla-pom-xml.md]: http://jdk.java.net/9/

File: en/openjdk-projects/kulla/kulla.md [hunk 8]
```diff
@@ -11,8 +11,8 @@ https://bugs.openjdk.java.net/browse/JDK-8043364
 View or add your comments to the recent LJC hackday [feedback](https://docs.google.com/document/d/1b236MW-cliUrmSWyVkBBs460Inh5lLcLvRsOmGThzlg/edit?usp=sharing) by the attendees.
 
 **Download binaries**  
-JDK9 binaries: https://jdk9.java.net/download/<br/>
-Kulla.jar: [Adopt OpenJDK Cloudbees Build farm](https://adopt-openjdk.ci.cloudbees.com/view/OpenJDK/job/langtools-1.9-linux-x86_64-kulla-dev/lastSuccessfulBuild/artifact/)
+JDK9 binaries: https://jdk9.java.net/download/ [dead-link]<br/>
```
Comment (neomatrix369) [inline en/openjdk-projects/kulla/kulla.md]: http://jdk.java.net/9/

File: en/openjdk-projects/nashorn.md [hunk 9]
```diff
@@ -14,10 +14,10 @@ Code base (OpenJDK9): <br/> http://hg.openjdk.java.net/jdk9/jdk9/nashorn
 NodeJS port on Nashorn (NodeJS on the JVM).
 
 Project: <br/>
-https://avatar-js.java.net/  https://avatar.java.net/
+https://avatar-js.java.net/ [dead-link]  https://avatar.java.net/ [dead-link]
```
Comment (neomatrix369) [inline en/openjdk-projects/nashorn.md]: Search for alternative sites on google

File: en/openjdk-projects/nashorn.md [hunk 10]
```diff
@@ -14,10 +14,10 @@ Code base (OpenJDK9): <br/> http://hg.openjdk.java.net/jdk9/jdk9/nashorn
 NodeJS port on Nashorn (NodeJS on the JVM).
 
 Project: <br/>
-https://avatar-js.java.net/  https://avatar.java.net/
+https://avatar-js.java.net/ [dead-link]  https://avatar.java.net/ [dead-link] 
 
 Sources:  <br/>
-https://java.net/projects/avatar-js/sources
+https://java.net/projects/avatar-js/sources [dead-link]
```
Comment (neomatrix369) [inline en/openjdk-projects/nashorn.md]: the site might have moved elsewhere

**Comment by neomatrix369** [comment]
I have assigned this back to you to see if you can fix some of the dead links and then we can merge it - thanks for all the effort @allandequeiroz

File: en/openjdk-projects/nashorn.md (hunk 9 above)
Comment (allandequeiroz) [inline en/openjdk-projects/nashorn.md]: Hmm, this is looking more something to be removed, even at Oracle's links are broken, like
https://blogs.oracle.com/otn/project-avatar,-avatar-js,-and-nashorn-david-delabassee

And other links that I found, were to say that Oracle quit Avatar.

File: en/openjdk-projects/nashorn.md (hunk 10 above)
Comment (neomatrix369) [inline en/openjdk-projects/nashorn.md]: Wayback machine?

**Comment by neomatrix369** [comment]
Try to give an attempt with the wayback machine on some of the links and when exhausted let me know - we should also stop with it at somepoint and give up saying can;t replace the links.

**Comment by allandequeiroz** [comment]
Hi @neomatrix369 I'll be back to it, is on my list, I'm trying to fix another issue then I'll finish this.

**Comment by neomatrix369** [comment]
Thanks @allandequeiroz look forward to it.

**Comment by allandequeiroz** [comment]
Hi @neomatrix369, just updated with all the links that I managed to find out. Sunset pages have no anchors so, some links that previously were pointing to specific parts of the pages are now pointing to the same page, content is there but users will need to scroll.

**Comment by neomatrix369** [comment]
@allandequeiroz Any luck with wayback machine ?

**Comment by allandequeiroz** [comment]
Hey @neomatrix369 yes I've found some way to put the information there for our Avatar links and fixed another ones.
https://goo.gl/2agqpY

Shall we use wayback for the missing ones too?

**Comment by neomatrix369** [comment]
@allandequeiroz Yes wherever possible and where it isn't taking too much time to fix them - I'm think we should wind this activity quickly and leave the pending in a git issue to tackle for later.

File: en/adopt-openjdk-getting-started/what_is_openjdk.md [hunk 11]
```diff
@@ -13,4 +13,4 @@ Read all about it in this at:
 * [Recent blog post on Voxxed](https://www.voxxed.com/blog/2015/01/java-jvm-built-adopt-openjdk-answer/)
 * [#Twitter-Scale computing with #OpenJDK](https://t.co/f4GDsfBADv)
 * [Some more OpenJDK related articles on Voxxed](https://www.voxxed.com/?s=openjdk)
-* [A post on Java Advent Calendar](http://www.javaadvent.com/2014/12/the-java-ecosystem-my-top-5-highlights.html) [dead-link]
```
Comment (neomatrix369) [inline en/adopt-openjdk-getting-started/what_is_openjdk.md]: The JavaAdvent links will be back soon, there is some work done to reinstate them, for now this is good

**Comment by allandequeiroz** [comment]
All right @neomatrix369, I'll start.

File: en/adopt-openjdk-getting-started/what_is_openjdk.md (hunk 11 above)
Comment (allandequeiroz) [inline en/adopt-openjdk-getting-started/what_is_openjdk.md]: Let me know to put it back.

**Comment by allandequeiroz** [comment]
Done @neomatrix369, I'll open an issue with the ones unavailable on wayback.

**Comment by neomatrix369** [comment]
thanks @allandequeiroz 
pending issues to be resolved via https://github.com/AdoptOpenJDK/adoptopenjdk-getting-started-kit/issues/66

**Pull request merged**